### PR TITLE
TASK-53709: The new changed username isn't considered in chat application.

### PR DESCRIPTION
**Pull request by IlhemEssaadi**
ISSUES : When editing username, The new changed username isn't considered in chat application.
FIX : The problem is that when editing the firstname or the last name ,the new changes does not considered in chat application and it solved by adding the new displayname to the user object that will be updated in the mongo database.

### CODE REVIEW

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you add a little unit test to validate your modification ?
Thanks

File: component/portal/src/test/java/org/exoplatform/portal/rest/UserRestResourcesTest.java [hunk 1]
```diff
@@ -461,6 +461,20 @@ public void testUpdateUser() throws Exception {
...
+    data.put("firstName", USER_2);
+    data.put("password", "");
+    data.put("enabled", true);
+    data.put("email", email2);
+    response = getResponse("PUT", "/v1/users", data.toString());
+    assertNotNull(response);
+    assertNull(response.getEntity());
+    assertEquals(204, response.getStatus());
+    user2.setDisplayName(USER_2+" "+USER_2);
+    verify(userHandler, atMost(1)).saveUser(eq(user2), eq(true));
```
Comment (rdenarie) [inline component/portal/src/test/java/org/exoplatform/portal/rest/UserRestResourcesTest.java:476]: Can you explain what this line verify ?

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you fix the sonar feedback

**Comment by IlhemEssaadi** [comment]
> Can you fix the sonar feedback

We can not fix them, they ask to remove the Mockito eq function which is required to pass the arguments. when I tried to remove one of them I got : 
```org.mockito.exceptions.misusing.InvalidUseOfMatchersException: 
Invalid use of argument matchers!
2 matchers expected, 1 recorded:
-> at org.exoplatform.portal.rest.UserRestResourcesTest.testUpdateUser(UserRestResourcesTest.java:476)

This exception may occur if matchers are combined with raw values:
    //incorrect:
    someMethod(anyObject(), "raw String");
When using matchers, all arguments have to be provided by matchers.
For example:
    //correct:
    someMethod(anyObject(), eq("String by matcher"));
```

File: component/portal/src/test/java/org/exoplatform/portal/rest/UserRestResourcesTest.java (hunk 1 above)
Comment (IlhemEssaadi) [reply component/portal/src/test/java/org/exoplatform/portal/rest/UserRestResourcesTest.java:476]: This line verify the save of the new user object after the update.

**Comment by rdenarie** [comment]
And what if you put 
`verify(userHandler, atMost(1)).saveUser(user2, true);`

**Comment by IlhemEssaadi** [comment]
> And what if you put `verify(userHandler, atMost(1)).saveUser(user2, true);`

The same problem!

**Review by rdenarie — APPROVED** [review]

**Pull request merged**